### PR TITLE
clippy fixes

### PR DESCRIPTION
**Pull request by deepink-mas**

### CODE REVIEW

File: bin/node-template/node/src/service.rs [hunk 1]
```diff
@@ -35,6 +35,7 @@ pub(crate) type FullClient =
 type FullBackend = sc_service::TFullBackend<Block>;
 type FullSelectChain = sc_consensus::LongestChain<FullBackend, Block>;
 
+#[allow(clippy::type_complexity)]
```
Comment (ggwpez) [inline bin/node-template/node/src/service.rs:38]: Does the CI fail without these?

Comment (deepink-mas) [reply bin/node-template/node/src/service.rs:38]: Not as far as I know. This is merely to silence a warning when you or someone else runs "cargo clippy" on your node.

Comment (ggwpez) [reply bin/node-template/node/src/service.rs:38]: The [config.toml](https://github.com/paritytech/substrate/blob/72dd574acf805fda1dc1e64b7bbd510e12a81312/.cargo/config.toml) defines the clippy behaviour I dont think it makes sense to introduce `allow`s beyond that, which are not automatically checked/enforced. 
cc @bkchr

File: bin/node-template/node/src/service.rs [hunk 2]
```diff
@@ -100,6 +101,8 @@ pub fn new_partial(
 
 	let (grandpa_block_import, grandpa_link) = sc_consensus_grandpa::block_import(
 		client.clone(),
+		// false positive
+		#[allow(clippy::redundant_clone)]
```
Comment (bkchr) [inline bin/node-template/node/src/service.rs:105]: Should be solved properly.

File: bin/node-template/node/src/service.rs (hunk 1 above)
Comment (bkchr) [reply bin/node-template/node/src/service.rs:38]: What is the problem here? What complexity? Of the argument?

File: bin/node-template/node/src/service.rs (hunk 2 above)
Comment (deepink-mas) [reply bin/node-template/node/src/service.rs:105]: This comment (+ allow annotation) is to record that we run into a clippy issue here. If accepted, I can then easily open a clippy issue pointed at this comment. Once it is fixed on the clippy side we can get rid of this comment and annotation.

File: bin/node-template/node/src/service.rs (hunk 1 above)
Comment (deepink-mas) [reply bin/node-template/node/src/service.rs:38]: It's the last type argument to `sc_service::PartialComponents`, namely:
```
		(
			sc_consensus_grandpa::GrandpaBlockImport<
				FullBackend,
				Block,
				FullClient,
				FullSelectChain,
			>,
			sc_consensus_grandpa::LinkHalf<Block, FullClient, FullSelectChain>,
			Option<Telemetry>,
		)
```

File: bin/node-template/node/src/service.rs [hunk 3]
```diff
@@ -100,6 +101,8 @@ pub fn new_partial(
...
 		&(client.clone() as Arc<_>),
```
Comment (bkchr) [inline bin/node-template/node/src/service.rs:106]:
```suggestion
		&client,
```

This is what I meant with "should be solved properly". This should make clippy stop complaining.

Comment (deepink-mas) [reply bin/node-template/node/src/service.rs:106]: Ah, I think I tried `&(client as Arc<_>),` which did not work, after which I assumed it was a false positive. Did you cargo check that suggestion?

Comment (bkchr) [reply bin/node-template/node/src/service.rs:106]: Yes

**Pull request closed without merging**